### PR TITLE
Uri postfix

### PR DESCRIPTION
**Pull request by MoMannn**
@typwrtr Requesting spell check.

### CODE REVIEW

File: docs/api/ethereum.md [hunk 1]
```diff
@@ -2446,14 +2449,15 @@ const mutation = await ledger.revokeAsset(assetId);
 An `asynchronous` class instance `function` which updates ledger data.
 
 ::: warning
-You need `UPDATE_URI_BASE` ledger ability to update ledger's `uriBase` property.
+You need `UPDATE_URI` ledger ability to update ledger's `uriPrefix` and `uriPostifx` properties.
```
Comment (typwrtr) [inline docs/api/ethereum.md]: You need `UPDATE_URI` ledger ability to update ledger's `uriPrefix` and `uriPostfix` properties.

File: docs/api/ethereum.md [hunk 2]
```diff
@@ -2446,14 +2449,15 @@ const mutation = await ledger.revokeAsset(assetId);
...
 :::
 
 **Arguments:**
 
 | Argument | Description
 |-|-
-| recipe.uriBase | [required] A `string` representing ledger URI base property.
+| recipe.uriPrefix | [required] A `string` representing ledger URI uriPrefix property.
+| recipe.uriPostifx | [required] A `string` representing ledger URI uriPostifx property.
```
Comment (typwrtr) [inline docs/api/ethereum.md]: | recipe.uriPostfix | [required] A `string` representing ledger URI uriPostfix property.

File: docs/api/wanchain.md [hunk 3]
```diff
@@ -1353,14 +1356,15 @@ const mutation = await ledger.revokeAsset(assetId);
 An `asynchronous` class instance `function` which updates ledger data.
 
 ::: warning
-You need `UPDATE_URI_BASE` ledger ability to update ledger's `uriBase` property.
+You need `UPDATE_URI` ledger ability to update ledger's `uriPrefix` and `uriPostifx` properties.
```
Comment (typwrtr) [inline docs/api/wanchain.md]: You need `UPDATE_URI` ledger ability to update ledger's `uriPrefix` and `uriPostfix` properties.

File: docs/api/wanchain.md [hunk 4]
```diff
@@ -1353,14 +1356,15 @@ const mutation = await ledger.revokeAsset(assetId);
...
 :::
 
 **Arguments:**
 
 | Argument | Description
 |-|-
-| recipe.uriBase | [required] A `string` representing ledger URI base property.
+| recipe.uriPrefix | [required] A `string` representing ledger URI uriPrefix property.
+| recipe.uriPostifx | [required] A `string` representing ledger URI uriPostifx property.
```
Comment (typwrtr) [inline docs/api/wanchain.md]: | recipe.uriPostfix | [required] A `string` representing ledger URI uriPostfix property.

File: packages/0xcert-ethereum-erc721-contracts/src/contracts/nf-token-metadata-enumerable.sol [hunk 5]
```diff
@@ -48,9 +48,14 @@ contract NFTokenMetadataEnumerable is
...
 
   /**
-   * @dev URI base for NFT metadata. NFT URI is made from base + NFT id.
+   * @dev URI prefix for NFT metadata. NFT URI is made from prefix + NFT id + postfix.
    */
-  string public uriBase;
+  string public uriPrefix;
+
+  /**
+   * @dev URI postfix for NFT metadata. NFT URI is made from prefix + NFT id + postfix.
```
Comment (typwrtr) [inline packages/0xcert-ethereum-erc721-contracts/src/contracts/nf-token-metadata-enumerable.sol]: * @dev URI postfix for NFT metadata. NFT URI is made from prefix + NFT ID + postfix.

File: packages/0xcert-ethereum-erc721-contracts/src/contracts/nf-token-metadata-enumerable.sol [hunk 6]
```diff
@@ -132,7 +137,8 @@ contract NFTokenMetadataEnumerable is
 
   /**
    * @dev Contract constructor.
-   * @notice When implementing this contract don't forget to set nftName, nftSymbol and uriBase.
+   * @notice When implementing this contract don't forget to set nftName, nftSymbol, uriPrefix and
```
Comment (typwrtr) [inline packages/0xcert-ethereum-erc721-contracts/src/contracts/nf-token-metadata-enumerable.sol]: * @notice When implementing this contract, don't forget to set nftName, nftSymbol, uriPrefix and

File: packages/0xcert-ethereum-erc721-contracts/src/contracts/nf-token-metadata.sol [hunk 7]
```diff
@@ -45,9 +45,14 @@ contract NFTokenMetadata is
   string internal nftSymbol;
 
   /**
-   * @dev URI base for NFT metadata. NFT URI is made from base + NFT id.
+   * @dev URI prefix for NFT metadata. NFT URI is made from prefix + NFT id + postfix.
```
Comment (typwrtr) [inline packages/0xcert-ethereum-erc721-contracts/src/contracts/nf-token-metadata.sol]: * @dev URI prefix for NFT metadata. NFT URI is made from prefix + NFT ID + postfix.

File: packages/0xcert-ethereum-erc721-contracts/src/contracts/nf-token-metadata.sol [hunk 8]
```diff
@@ -45,9 +45,14 @@ contract NFTokenMetadata is
...
    */
-  string public uriBase;
+  string public uriPrefix;
+
+  /**
+   * @dev URI postfix for NFT metadata. NFT URI is made from prefix + NFT id + postfix.
```
Comment (typwrtr) [inline packages/0xcert-ethereum-erc721-contracts/src/contracts/nf-token-metadata.sol]: * @dev URI postfix for NFT metadata. NFT URI is made from prefix + NFT ID + postfix.

File: packages/0xcert-ethereum-erc721-contracts/src/contracts/nf-token-metadata.sol [hunk 9]
```diff
@@ -114,7 +119,8 @@ contract NFTokenMetadata is
 
   /**
    * @dev Contract constructor.
-   * @notice When implementing this contract don't forget to set nftName, nftSymbol and uriBase.
+   * @notice When implementing this contract don't forget to set nftName, nftSymbol, uriPrefix and
```
Comment (typwrtr) [inline packages/0xcert-ethereum-erc721-contracts/src/contracts/nf-token-metadata.sol]: * @notice When implementing this contract, don't forget to set nftName, nftSymbol, uriPrefix and

File: packages/0xcert-scaffold/src/core/asset-ledger.ts [hunk 10]
```diff
@@ -212,7 +212,14 @@ export interface AssetLedgerDeployRecipe {
    * Example: https://example.com/id/
    * Asset 1 URI will become: https://example.com/id/1
    */
-  uriBase: string;
+  uriPrefix: string;
+
+  /**
+   * Uri postfix for metadata URI-s. After uriPrefix and assetId postfix is automatically appended for each asset.
```
Comment (typwrtr) [inline packages/0xcert-scaffold/src/core/asset-ledger.ts]: * URI postfix for metadata URIs. After uriPrefix and assetId, postfix is automatically appended for each asset.

File: packages/0xcert-scaffold/src/core/asset-ledger.ts [hunk 11]
```diff
@@ -266,7 +273,14 @@ export interface AssetLedgerInfo {
...
```
Comment (typwrtr) [inline packages/0xcert-scaffold/src/core/asset-ledger.ts]: * URI postfix for metadata URIs. After uriPrefix and assetId, postfix is automatically appended for each asset.

File: packages/0xcert-scaffold/src/core/asset-ledger.ts [hunk 12]
```diff
@@ -343,9 +357,16 @@ export interface AssetLedgerObjectUpdateRecipe {
 export interface AssetLedgerUpdateRecipe {
 
   /**
-   * Uri base for metadata URI-s. At the end of the base the assetId is automatically appended foo each asset.
+   * Uri base for metadata URI-s. At the end of the base the assetId is automatically appended for each asset.
```
Comment (typwrtr) [inline packages/0xcert-scaffold/src/core/asset-ledger.ts]: * URI base for metadata URIs. At the end of the base, the assetId is automatically appended for each asset.

File: packages/0xcert-scaffold/src/core/asset-ledger.ts [hunk 13]
```diff
@@ -343,9 +357,16 @@ export interface AssetLedgerObjectUpdateRecipe {
...
    * Example: https://example.com/id/
-   * Asset 1 URI will become: https://example.com/id/1
+   * Asset 1 URI will become: https://example.com/id/1 + postfix
+   */
+  uriPrefix: string;
+
+  /**
+   * Uri postfix for metadata URI-s. After uriPrefix and assetId postfix is automatically appended for each asset.
```
Comment (typwrtr) [inline packages/0xcert-scaffold/src/core/asset-ledger.ts]: * URI postfix for metadata URIs. After uriPrefix and assetId, postfix is automatically appended for each asset.

File: packages/0xcert-scaffold/src/core/gateway.ts [hunk 14]
```diff
@@ -309,11 +309,18 @@ export interface AssetLedgerData {
   symbol: string;
 
   /**
-   * Uri base for metadata URI-s. At the end of the base the assetId is automatically appended foo each asset.
+   * Uri base for metadata URI-s. At the end of the base the assetId is automatically appended for each asset.
```
Comment (typwrtr) [inline packages/0xcert-scaffold/src/core/gateway.ts]: * URI base for metadata URIs. At the end of the base, the assetId is automatically appended for each asset.

File: packages/0xcert-scaffold/src/core/gateway.ts [hunk 15]
```diff
@@ -309,11 +309,18 @@ export interface AssetLedgerData {
...
    * Example: https://example.com/id/
-   * Asset 1 URI will become: https://example.com/id/1
+   * Asset 1 URI will become: https://example.com/id/1 + postfix
    */
-  uriBase: string;
+  uriPrefix: string;
+
+  /**
+   * Uri postfix for metadata URI-s. After uriPrefix and assetId postfix is automatically appended for each asset.
```
Comment (typwrtr) [inline packages/0xcert-scaffold/src/core/gateway.ts]: * URI postfix for metadata URIs. After uriPrefix and assetId, postfix is automatically appended for each asset.

File: docs/api/ethereum.md [hunk 16]
```diff
@@ -1854,7 +1855,8 @@ const capabilities = [
 const recipe = {
     name: 'Math Course Certificate',
     symbol: 'MCC',
-    uriBase: 'http://domain.com/assets/',
+    uriPrefix: 'http://domain.com/assets/',
```
Comment (fulldecent) [inline docs/api/ethereum.md]: Please use example.com, as reserved by https://tools.ietf.org/html/rfc2606

File: docs/api/ethereum.md [hunk 17]
```diff
@@ -2464,7 +2468,8 @@ An instance of the same mutation class.
 ```ts
 // arbitrary data
 const recipe = {
-    uriBase: 'http://swapmarket.com/',
+    uriPrefix: 'http://swapmarket.com/',
```
Comment (fulldecent) [inline docs/api/ethereum.md]: https

File: docs/api/wanchain.md [hunk 18]
```diff
@@ -761,7 +762,8 @@ const capabilities = [
 const recipe = {
     name: 'Math Course Certificate',
     symbol: 'MCC',
-    uriBase: 'http://domain.com/assets/',
+    uriPrefix: 'http://domain.com/assets/',
```
Comment (fulldecent) [inline docs/api/wanchain.md]: example.com
https

File: docs/api/wanchain.md [hunk 19]
```diff
@@ -1371,7 +1375,8 @@ An instance of the same mutation class.
 ```ts
 // arbitrary data
 const recipe = {
-    uriBase: 'https://0xcert.com/asset/',
+    uriPrefix: 'http://swapmarket.com/',
```
Comment (fulldecent) [inline docs/api/wanchain.md]: https

File: docs/guide/asset-management.md [hunk 20]
```diff
@@ -54,7 +55,7 @@ We can now perform `query` and `mutation` requests on this newly deployed asset
 
 ```ts
 const assetLedgerInfo = await assetLedger.getInfo();
-//=> { name: 'Certificate', symbol: 'CERT', uriBase: 'http://domain.com', schemaId: '239423' }
+//=> { name: 'Certificate', symbol: 'CERT', uriPrefix: 'http://domain.com', uriPostfix: '.json', schemaId: '239423' }
```
Comment (fulldecent) [inline docs/guide/asset-management.md]: https
example.com

File: packages/0xcert-ethereum-asset-ledger/src/tests/core/ledger/deploy-static-method.test.ts [hunk 21]
```diff
@@ -31,7 +33,8 @@ spec.test('deploys new asset ledger', async (ctx) => {
   const info = {
     name: 'Foo',
     symbol: 'Bar',
-    uriBase: 'http://foo.bar',
+    uriPrefix: 'http://foo.bar',
```
Comment (fulldecent) [inline packages/0xcert-ethereum-asset-ledger/src/tests/core/ledger/deploy-static-method.test.ts]: please use any reserved domain at https://tools.ietf.org/html/rfc2606

File: packages/0xcert-ethereum-asset-ledger/src/tests/core/ledger/update-instance-method.test.ts [hunk 22]
```diff
@@ -28,13 +28,15 @@ spec.before(async (stage) => {
   stage.set('ledger', new AssetLedger(provider, ledgerId));
 });
 
-spec.test('update uri base', async (ctx) => {
+spec.test('update uri', async (ctx) => {
   const xcert = ctx.get('protocol').xcert;
   const ledger = ctx.get('ledger');
   await ledger.update({
-    uriBase: 'http://new.com/',
+    uriPrefix: 'http://new.com/',
```
Comment (fulldecent) [inline packages/0xcert-ethereum-asset-ledger/src/tests/core/ledger/update-instance-method.test.ts]: use domain from https://tools.ietf.org/html/rfc2606

File: packages/0xcert-ethereum-gateway-contracts/src/tests/xcert-deploy-gateway/cancel-instance-method.test.ts [hunk 23]
```diff
@@ -82,7 +82,8 @@ spec.test('cancels an deploy', async (ctx) => {
     deployData: {
       name: 'test',
       symbol: 'TST',
-      uriBase: 'https://base.com/',
+      uriPrefix: 'https://base.com/',
```
Comment (fulldecent) [inline packages/0xcert-ethereum-gateway-contracts/src/tests/xcert-deploy-gateway/cancel-instance-method.test.ts]: Chernobyl?

https://tools.ietf.org/html/rfc2606

File: packages/0xcert-ethereum-gateway/src/tests/core/gateway/asset-ledger-deploy-order/cancel-instance-method.test.ts [hunk 24]
```diff
@@ -71,7 +71,8 @@ spec.test('marks xcertDeployGateway deploy as canceled on the network which prev
     assetLedgerData: {
       name: 'test',
       symbol: 'TST',
-      uriBase: 'https://base.com/',
+      uriPrefix: 'https://base.com/',
```
Comment (fulldecent) [inline packages/0xcert-ethereum-gateway/src/tests/core/gateway/asset-ledger-deploy-order/cancel-instance-method.test.ts]: https://tools.ietf.org/html/rfc2606

File: packages/0xcert-ethereum-gateway/src/tests/core/gateway/asset-ledger-deploy-order/get-order-data-claim-method-instance.test.ts [hunk 25]
```diff
@@ -52,7 +52,8 @@ spec.test('check if deploy data claim equals locally created one', async (ctx) =
     assetLedgerData: {
       name: 'test',
       symbol: 'TST',
-      uriBase: 'https://base.com/',
+      uriPrefix: 'https://base.com/',
```
Comment (fulldecent) [inline packages/0xcert-ethereum-gateway/src/tests/core/gateway/asset-ledger-deploy-order/get-order-data-claim-method-instance.test.ts]: https://tools.ietf.org/html/rfc2606

File: packages/0xcert-ethereum-gateway/src/tests/core/gateway/asset-ledger-deploy-order/is-valid-signature-instance-method.test.ts [hunk 26]
```diff
@@ -51,7 +51,8 @@ spec.test('check if signature is valid', async (ctx) => {
     assetLedgerData: {
       name: 'test',
       symbol: 'TST',
-      uriBase: 'https://base.com/',
+      uriPrefix: 'https://base.com/',
```
Comment (fulldecent) [inline packages/0xcert-ethereum-gateway/src/tests/core/gateway/asset-ledger-deploy-order/is-valid-signature-instance-method.test.ts]: https://tools.ietf.org/html/rfc2606

File: packages/0xcert-ethereum-gateway/src/tests/core/gateway/asset-ledger-deploy-order/perform-instance-method.test.ts [hunk 27]
```diff
@@ -75,7 +75,8 @@ spec.test('submits gateway asset ledger deploy order to the network which execut
     assetLedgerData: {
       name: 'test',
       symbol: 'TST',
-      uriBase: 'https://base.com/',
+      uriPrefix: 'https://base.com/',
```
Comment (fulldecent) [inline packages/0xcert-ethereum-gateway/src/tests/core/gateway/asset-ledger-deploy-order/perform-instance-method.test.ts]: https://tools.ietf.org/html/rfc2606

File: packages/0xcert-ethereum-gateway/src/tests/core/gateway/asset-ledger-deploy-order/perform-instance-method.test.ts [hunk 28]
```diff
@@ -121,7 +122,8 @@ spec.test('submits dynamic asset ledger deploy order to the network which execut
...
```
Comment (fulldecent) [inline packages/0xcert-ethereum-gateway/src/tests/core/gateway/asset-ledger-deploy-order/perform-instance-method.test.ts]: https://tools.ietf.org/html/rfc2606

File: packages/0xcert-ethereum-gateway/src/tests/core/gateway/asset-ledger-deploy-order/perform-instance-method.test.ts [hunk 29]
```diff
@@ -166,7 +168,8 @@ spec.test('handles fixed deploy with dynamic token transfer receiver', async (ct
...
```
Comment (fulldecent) [inline packages/0xcert-ethereum-gateway/src/tests/core/gateway/asset-ledger-deploy-order/perform-instance-method.test.ts]: https://tools.ietf.org/html/rfc2606

File: packages/0xcert-vue-example/pages/index.vue [hunk 30]
```diff
@@ -99,7 +99,7 @@ export default {
       const mutation = await this.$0xcert.deployAssetLedger({
         name: 'fundin',
         symbol: 'fundin',
-        uriBase: 'http://www.fundin.us',
+        uriPrefix: 'http://www.fundin.us',
```
Comment (fulldecent) [inline packages/0xcert-vue-example/pages/index.vue]: https://tools.ietf.org/html/rfc2606

**Pull request merged**